### DISCUSSION TITLE
Is there a way to monkey patch a plugin?

### DISCUSSION THREAD

**Discussion by m00qek** [wbthomason/packer.nvim#611] [Q&A]
Hello everyone,

Do you know if is there any way to tell a plugin to use my custom version of `vim.api.nvim_set_buf_lines`? I was thinking about doing a [monkey patch](https://en.wikipedia.org/wiki/Monkey_patch)... don't know how to do it though. 
Doing this would be awesome to test some changes on the fly ✈️ 

Thanks!

**Comment by wbthomason** [comment]
This really depends on how the plugin is defined and what you're trying to override, but it's pretty easy to override things in Lua just by redefining them. For example, to override `nvim_set_buf_lines`, you can simply set `vim.api.nvim_set_buf_lines = function(...) ... end`.
